tests/test/export/test.sh

```diff
@@ -0,0 +1,145 @@
+#!/bin/bash
+
+. /usr/share/beakerlib/beakerlib.sh || exit 1
+
+function verify_output
+{
+    typeset output_type=${1?"*** yaml OR dict"}
+    shift
+    typeset fields="$*"
+
+    typeset field=""
+    for field in $fields; do
+        if [[ "$output_type" == "yaml" ]]; then
+            rlAssertGrep "$field:"   $rlRun_LOG
+        else
+            rlAssertGrep "'$field':" $rlRun_LOG
+        fi
+    done
+}
+
+
+rlJournalStart
+    rlPhaseStartSetup
+        rlRun "pushd data"
+        rlRun "set -o pipefail"
+    rlPhaseEnd
+
+    tnames=""
+    tnames+=" /tests/test/export/data/tests/enabled/default"
+    tnames+=" /tests/test/export/data/tests/enabled/defined"
+    tnames+=" /tests/test/export/data/tests/enabled/disabled"
+    tnames+=" /tests/test/export/data/tests/tag/default"
+    tnames+=" /tests/test/export/data/tests/tag/defined"
+    tnames+=" /tests/test/export/data/tests/tag/empty"
+    tnames+=" /tests/test/export/data/tests/tier/default"
+    tnames+=" /tests/test/export/data/tests/tier/one"
+    tnames+=" /tests/test/export/data/tests/tier/two"
+
+    # 1 - (positive) format testing
+    fields=""
+    fields+=" summary"
+    fields+=" description"
+    fields+=" contact"
+    fields+=" component"
+    fields+=" test"
+    fields+=" path"
+    fields+=" framework"
+    fields+=" manual"
+    fields+=" require"
+    fields+=" recommend"
+    fields+=" environment"
+    fields+=" duration"
+    fields+=" order"
+    fields+=" result"
+    fields+=" tag"
+    fields+=" tier"
+    fields+=" link"
+
+    cmd="tmt test export ."
+    rlPhaseStartTest "$cmd"
+        rlRun -s "$cmd" 0 "Export test"
+        for tname in $tnames; do
+            rlAssertGrep "- name: $tname" $rlRun_LOG
+        done
+        verify_output "yaml" "$fields"
+    rlPhaseEnd
```

Comment:
This grepping of individual lines does not actually verify that a valid list of multiple tests is actually exported. Except for the missing `name` the lines above pass without any problem with the old broken implementation as well.

What about doing a real parsing to verify that the output it really a valid `yaml`? It could look like this:

```shell
rlRun -s "tmt tests export | ./parse.py"
```

```python
#!/usr/bin/python3

import sys
import ruamel.yaml

yaml = ruamel.yaml.YAML()
tests = yaml.load(sys.stdin)

print(tests[0]["name"])
assert tests[0]["name"] == "/test/one"

print(tests[1]["name"])
assert tests[0]["name"] == "/test/two"

print(tests[2]["name"])
assert tests[0]["name"] == "/test/three"
```

This would actually catch the problem with the original output:

```
found duplicate key "summary" with value "This test is enabled explicitly" (original value: "This test is enabled by default")
  in "<stdin>", line 19, column 1
```

What do you think?

Reply:
@psss, nice suggestion! I have created the [parse.py][01] file, please help to review again, thanks!

[01]: https://github.com/idorax/tmt/blob/dev.huanli.20220321.1091.export/tests/test/export/parse.py